src/yz_kv.erl

```diff
@@ -190,8 +190,20 @@ has_indexes(RemoteNode) ->
         _ -> false
     end.
 
-index(Obj, Reason, P) ->
-    case yokozuna:is_enabled(index) andalso ?YZ_ENABLED of
+-spec is_enabled() -> boolean().
+is_enabled() ->
+    yokozuna:is_enabled(index) andalso ?YZ_ENABLED.
+
+index(B, K, Obj, Reason, P) when is_binary(Obj) ->
```

Comment:
maybe we can add a dialyzer spec to index now as well and a comment about allowing for Bucket and Key to pass through and why the latter version treats it as unnecessary.